core/src/main/scala/chisel3/Layer.scala

```diff
@@ -19,35 +19,48 @@ object layer {
   /** Enumerations of different optional layer conventions.  A layer convention
     * says how a given layer should be lowered to Verilog.
     */
+  @deprecated
   object Convention {
     sealed trait Type
 
-    /** Internal type used as the parent of all layers. */
-    private[chisel3] case object Root extends Type
-
     /** The layer should be lowered to a SystemVerilog `bind`. */
-    case object Bind extends Type
+    final case object Bind extends Type
   }
 
   sealed trait OutputDirBehavior
   final case object DefaultOutputDir extends OutputDirBehavior
   final case object NoOutputDir extends OutputDirBehavior
   final case class CustomOutputDir(path: Path) extends OutputDirBehavior
 
+  sealed abstract class LayerConfig
```

Comment:
Nit: a `trait` is slightly cleaner here.

```suggestion
  sealed trait LayerConfig
```